src/tigerbeetle/inspect.zig

```diff
@@ -0,0 +1,1100 @@
+
+pub fn main(allocator: std.mem.Allocator, cli_args: *const cli.Command.Inspect) !void {
+    var stdout_buffer = std.io.bufferedWriter(std.io.getStdOut().writer());
+
+    const stdout = stdout_buffer.writer();
+
+    var inspector = try Inspector.init(allocator, cli_args.path);
+    defer inspector.deinit();
+
+    switch (cli_args.query) {
+        .superblock => try inspector.inspect_superblock(stdout),
+        .wal => |args| {
+            if (args.slot) |slot| {
+                if (slot >= constants.journal_slot_count) {
+                    return fatal(
+                        "--slot: slot exceeds {}",
+                        .{constants.journal_slot_count - 1},
+                    );
+                }
+                try inspector.inspect_wal_slot(stdout, slot);
+            } else {
+                try inspector.inspect_wal(stdout);
+            }
+        },
+        .replies => |args| {
+            if (args.slot) |slot| {
+                if (slot >= constants.clients_max) {
+                    return fatal("--slot: slot exceeds {}", .{constants.clients_max - 1});
+                }
+                try inspector.inspect_replies_slot(stdout, args.superblock_copy, slot);
+            } else {
+                try inspector.inspect_replies(stdout, args.superblock_copy);
+            }
+        },
+        .grid => |args| {
+            if (args.superblock_copy >= constants.superblock_copies) {
+                return fatal(
+                    "--superblock-copy: copy exceeds {}\n",
+                    .{constants.superblock_copies - 1},
+                );
+            }
+
+            if (args.block) |address| {
+                try inspector.inspect_grid_block(stdout, address);
+            } else {
+                try inspector.inspect_grid(stdout, args.superblock_copy);
+            }
+        },
+        .manifest => |args| {
+            if (args.superblock_copy >= constants.superblock_copies) {
+                return fatal(
+                    "--superblock-copy: copy exceeds {}\n",
+                    .{constants.superblock_copies - 1},
+                );
+            }
+
+            try inspector.inspect_manifest(stdout, args.superblock_copy);
+        },
+        .tables => |args| {
+            const tree_id = parse_tree_id(args.tree) orelse {
+                return fatal("--tree: invalid tree name/id: {s}\n", .{args.tree});
+            };
+            try inspector.inspect_tables(stdout, args.superblock_copy, .{
+                .tree_id = tree_id,
+                .level = args.level,
+            });
+        },
+    }
+
+    try stdout_buffer.flush();
+}
+
+const Inspector = struct {
+    allocator: std.mem.Allocator,
+    dir_fd: std.posix.fd_t,
+    fd: std.posix.fd_t,
+    io: vsr.io.IO,
+    storage: Storage,
+
+    busy: bool = false,
+    read: Storage.Read = undefined,
+
+    fn init(allocator: std.mem.Allocator, path: []const u8) !*Inspector {
```

Comment:
As this returns a pointer, this could use `create/destroy` rather than `init/deinit`. Not sure if this naming convention is wortth pursuing though!